frontend/src/actions/WindowActions.js

```diff
@@ -521,38 +528,64 @@ export function deselectTableItems(ids, windowType, viewId) {
 
 // THUNK ACTIONS
 
-export function initTabs(layout, windowType, docId, isModal) {
-  return async (dispatch) => {
-    const requests = [];
-    const tabTmp = {};
-
-    if (layout) {
-      layout.map((tab, index) => {
-        tabTmp[tab.tabId] = {};
-
-        if ((tab.tabId && index === 0) || !tab.queryOnActivate) {
-          requests.push(getTab(tab.tabId, windowType, docId));
-        }
-      });
+// TODO: Just a quick thunk action creator to test Tables reducer
+// but looks like this can actually replace the `initTabs`.
+/*
+ * @method fetchTab
+ * @summary Action creator for fetching single tab's rows
+ */
+export function fetchTab(tabId, windowType, docId) {
```

Comment:
Same as above. I don't see the advantage especially that we're still using `windowId` in some places so it might turn out this will cause some unexpected issues.